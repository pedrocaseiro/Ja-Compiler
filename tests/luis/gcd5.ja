class gcd5 {
	public static void main(int a, double d) {

	}

	public static void main(int a, int d) {

	}

	public static void caller() {
		main(1,1);
		System.out.println(a);
		if(!a) {

		}
		double a;
		System.out.println(a);
	}

	public static int a;
}