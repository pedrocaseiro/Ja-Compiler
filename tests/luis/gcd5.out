Line 13, col 6: Operator ! cannot be applied to type int
===== Class gcd5 Symbol Table =====
main	(int,double)	void
main	(int,int)	void
caller	()	void
a		int

===== Method main(int,double) Symbol Table =====
return		void
a		int	param
d		double	param

===== Method main(int,int) Symbol Table =====
return		void
a		int	param
d		int	param

===== Method caller() Symbol Table =====
return		void
a		double

Program
..Id(gcd5)
..MethodDecl
....MethodHeader
......Void
......Id(main)
......MethodParams
........ParamDecl
..........Int
..........Id(a)
........ParamDecl
..........Double
..........Id(d)
....MethodBody
..MethodDecl
....MethodHeader
......Void
......Id(main)
......MethodParams
........ParamDecl
..........Int
..........Id(a)
........ParamDecl
..........Int
..........Id(d)
....MethodBody
..MethodDecl
....MethodHeader
......Void
......Id(caller)
......MethodParams
....MethodBody
......Call - void
........Id(main) - (int,int)
........DecLit(1) - int
........DecLit(1) - int
......Print
........Id(a) - int
......If
........Not - boolean
..........Id(a) - int
........Block
........Block
......VarDecl
........Double
........Id(a)
......Print
........Id(a) - double
..FieldDecl
....Int
....Id(a)
